crates/polars-io/src/cloud/options.rs

```diff
@@ -120,16 +120,14 @@ fn parsed_untyped_config<T, I: IntoIterator<Item = (impl AsRef<str>, impl Into<S
 where
     T: FromStr + Eq + std::hash::Hash,
 {
-    config
+    Ok(config
         .into_iter()
-        .map(|(key, val)| {
-            T::from_str(key.as_ref())
-                .map_err(
-                    |_| polars_err!(ComputeError: "unknown configuration key: {}", key.as_ref()),
-                )
+        .filter_map(|(key, val)| {
```

Comment:
Now invalid keys are silently ignored? That doesn't seem right.

Also, doesn't AWS expect some upper case keys. We are lowering them now.

Reply:
I am doing it explicitly. In delta-rs for example we use quite some custom storage_options for our internal log/objectstore handling.

So when you do a read_delta and you pass those additional options from your DeltaTable object into pola-rs, it will error out, see this comment https://github.com/pola-rs/polars/pull/19103#issuecomment-2483144270

Reply:
Right, can you add a comment that some keys are read upstream and may silently be ignored there.

Reply:
Donee